src/com/ms/silverking/io/IOUtil.java

```diff
@@ -7,20 +7,20 @@
 import java.util.Collection;
 
 public class IOUtil {
-    public static <T> void writeAsLines(File outFile, Collection<T> c) throws IOException {
-        writeAsLines(new FileOutputStream(outFile), c, true);
-    }
-    
-    public static <T> void writeAsLines(OutputStream out, Collection<T> c, boolean close) throws IOException {
-        try {
-            for (T element : c) {
-                out.write(element.toString().getBytes());
-                out.write('\n');
-            }
-        } finally {
-            if (close) {
-                out.close();
-            }
-        }
+  public static <T> void writeAsLines(File outFile, Collection<T> c) throws IOException {
+    writeAsLines(new FileOutputStream(outFile), c, true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileOutputStream*. The resource is referenced in statements at the following line: 11. The resource closure statement is at line: 22. There are other execution paths that do not contain closure statements, for example, when an else-branch is taken at line 21. Either a) close the object returned by *FileOutputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileOutputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).